pkg/apply/csaupgrade.go

```diff
@@ -0,0 +1,250 @@
+package apply
+
+import (
+	"bytes"
+	"fmt"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"reflect"
+	"sigs.k8s.io/structured-merge-diff/v4/fieldpath"
+)
+
+/*
+this sub-package is taken from k8s.io/client-go/util/csaupgrade
```

Comment:
I wouldn't say sub-package here to avoid confusion, the way you did it is just this single file.

Reply:
removed